### PR TITLE
Feature/add/box shadow to bar chart rod data

### PR DESCRIPTION
**Pull request by SomeoneAndNoone**
https://github.com/imaNNeoFighT/fl_chart/issues/924

### CODE REVIEW

File: lib/src/chart/bar_chart/bar_chart_data.dart [hunk 1]
```diff
@@ -295,6 +295,9 @@ class BarChartRodData with EquatableMixin {
   /// If you want to have a border for rod, set this value.
   final BorderSide borderSide;
 
+  /// adds shadow to the rod
+  final BoxShadow? boxShadow;
```
Comment (imaNNeoFighT) [inline lib/src/chart/bar_chart/bar_chart_data.dart:299]: Please add this property in `copyWith()` and `lerp()` functions below.

File: lib/src/chart/bar_chart/bar_chart_painter.dart [hunk 2]
```diff
@@ -349,6 +352,8 @@ class BarChartPainter extends AxisChartPainter<BarChartData> {
 
               _barPaint.color = stackItem.color;
               canvasWrapper.save();
+              // _paintShadow(
```
Comment (imaNNeoFighT) [inline lib/src/chart/bar_chart/bar_chart_painter.dart:355]: Remove commented codes.

File: lib/src/chart/bar_chart/bar_chart_painter.dart [hunk 3]
```diff
@@ -371,6 +376,23 @@ class BarChartPainter extends AxisChartPainter<BarChartData> {
...
+
+    final Rect bounds = Rect.fromLTRB(
+      barRRect.left,
+      barRRect.top,
+      barRRect.right,
+      barRRect.bottom,
+    ).shift(boxShadow.offset).inflate(boxShadow.spreadRadius);
+
+    final Rect newRect =
+        bounds.shift(boxShadow.offset).inflate(boxShadow.spreadRadius);
```
Comment (imaNNeoFighT) [inline lib/src/chart/bar_chart/bar_chart_painter.dart:391]: Why it has shifted and inflated two times? (check line 388)

File: lib/src/chart/bar_chart/bar_chart_painter.dart [hunk 4]
```diff
@@ -371,6 +376,23 @@ class BarChartPainter extends AxisChartPainter<BarChartData> {
     }
   }
 
+  void _paintShadow(
```
Comment (imaNNeoFighT) [inline lib/src/chart/bar_chart/bar_chart_painter.dart:379]: Don't forget to write unit-test for this.

Comment (SomeoneAndNoone) [reply lib/src/chart/bar_chart/bar_chart_painter.dart]: @imaNNeoFighT  unfortunately I don't have any experience for this. Could you help me on it 🙂?

Comment (imaNNeoFighT) [reply lib/src/chart/bar_chart/bar_chart_painter.dart]: Yes, for sure. I'm here to help.
I will give you some clues.

**Importance:**
First of all, you need to understand the unit-test importance. 
When we write logic (any logic, here we are painting something), we need to write a test function which verifies that logic is working correctly. 
All of these test functions will run automatically when anyone changes the code. In this way, we can realize if some changes on this logic break it or not automatically.

**Where to start:**
We have a [CONTRIBUTING](https://github.com/imaNNeoFighT/fl_chart/blob/master/CONTRIBUTING.md#test-coverage-unit-tests) doc which you can read about our unit tests system.

Then I suggest you read some of the unit tests under the `/tests` directory (start with simple functions)

Cheers!

**Comment by imaNNeoFighT** [comment]
Our branching approach changed due to the problem you have faced. (Read more about it here #938).
Sorry about that I had to do that.
Now you need to make your branch from `master`.

Follow this guide to re-create your PR:
1. Checkout to `master` branch and pull it.
2. Create a new branch
3. Cherry-pick your commits from this old branch into your new branch.
4. Then push it and make a new PR. 
5. Mention the old PR in your new PR.

**Pull request closed without merging**